### PR TITLE
shields: nrf7002_ek: Add support for nrf7000 and nrf7001

### PR DESCRIPTION
**Pull request by rado17**
Add support for nrf7000 and nrf7001 EK devices.

### CODE REVIEW

File: samples/nrf9160/modem_shell/sample.yaml [hunk 1]
```diff
@@ -40,14 +40,42 @@ tests:
...
+      - nrf9160dk_nrf9160_ns
+    platform_allow: nrf9160dk_nrf9160_ns
+    extra_args: SHIELD=nrf7002ek_nrf7000 OVERLAY_CONFIG="overlay-nrf7002ek-wifi-scan-only.conf;overlay-debug.conf" DTC_OVERLAY_FILE="nrf9160dk_with_nrf7002ek.overlay"
+    tags: ci_build
+  sample.nrf9160.modem_shell.nrf7001ek_wifi-debug:
+    build_only: true
+    integration_platforms:
+      - nrf9160dk_nrf9160_ns
+    platform_allow: nrf9160dk_nrf9160_ns
+    extra_args: SHIELD=nrf7002ek_nrf7001 OVERLAY_CONFIG="overlay-nrf7002ek-wifi-scan-only.conf;overlay-debug.conf" DTC_OVERLAY_FILE="nrf9160dk_with_nrf7002ek.overlay"
```
[the remark this answers is not in the document]
Comment (tokangas) [reply samples/nrf9160/modem_shell/sample.yaml:78]: I agree, that's something to consider because there's a lot of nrf700x configurations now.

Maybe we should also change the naming of the related `modem_shell` overlays? But that can be done outside this PR.

File: boards/shields/nrf7002_ek/Kconfig.shield [hunk 2]
```diff
@@ -4,5 +4,14 @@
...
+DT_COMPAT_NORDIC_NRF7002_SPI := nordic,nrf7002-spi
+
 config SHIELD_NRF7002_EK
 	def_bool $(shields_list_contains,nrf7002_ek)
+
+config NRF7002_ON_QSPI
+	def_bool $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF7002_QSPI))
+
+config NRF7002_ON_SPI
+	def_bool $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF7002_SPI))
```
Comment (tokangas) [inline boards/shields/nrf7002_ek/Kconfig.shield:17]: This file is missing a linefeed at the end. Not sure it makes a difference, though.

File: boards/shields/nrf7002ek_nrf7001/Kconfig.shield [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+DT_COMPAT_NORDIC_NRF7002_SPI := nordic,nrf7001-spi
+
+config SHIELD_NRF7002EK_NRF7001
+	def_bool $(shields_list_contains,nrf7002ek_nrf7001)
+
+config NRF7001_ON_QSPI
+	def_bool $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF7002_QSPI))
+
+config NRF7001_ON_SPI
+	def_bool $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF7002_SPI))
```
Comment (tokangas) [inline boards/shields/nrf7002ek_nrf7001/Kconfig.shield:17]: Missing linefeed.

File: boards/shields/nrf7002ek_nrf7002/Kconfig.shield [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+DT_COMPAT_NORDIC_NRF7002_SPI := nordic,nrf7002-spi
+
+config SHIELD_NRF7002EK_NRF7002
+	def_bool $(shields_list_contains,nrf7002ek_nrf7002)
+
+config NRF7002_ON_QSPI
+	def_bool $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF7002_QSPI))
+
+config NRF7002_ON_SPI
+	def_bool $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF7002_SPI))
```
Comment (tokangas) [inline boards/shields/nrf7002ek_nrf7002/Kconfig.shield:17]: Missing linefeed.

File: samples/wifi/scan/sample.yaml [hunk 5]
```diff
@@ -11,9 +11,25 @@ tests:
...
+    platform_allow: nrf5340dk_nrf5340_cpuapp nrf52840dk_nrf52840
+    tags: ci_build
+  sample.nrf7001_eks.scan:
+    build_only: true
+    extra_args: SHIELD=nrf7002ek_nrf7001
+    integration_platforms:
+      - nrf5340dk_nrf5340_cpuapp
+      - nrf52840dk_nrf52840
+    platform_allow: nrf5340dk_nrf5340_cpuapp nrf52840dk_nrf52840
+    tags: ci_build
```
Comment (tokangas) [inline samples/wifi/scan/sample.yaml:35]: Missing linefeed.

**Review by tokangas — COMMENTED** [review]
A generic comment about the copyright in new files, the year could/should be updated to 2023.

File: boards/shields/nrf7002_ek/Kconfig.shield (hunk 2 above)
Comment (rado17) [reply boards/shields/nrf7002_ek/Kconfig.shield:17]: Added LF

File: samples/wifi/shell/sample.yaml [hunk 6]
```diff
@@ -47,7 +77,21 @@ tests:
     tags: ci_build
   sample.nrf7002.shell.scan_only_9160:
     build_only: true
-    extra_args: OVERLAY_CONFIG=overlay-scan-only.conf SHIELD=nrf7002_ek
+    extra_args: OVERLAY_CONFIG=overlay-scan-only.conf SHIELD=nrf7002ek_nrf7002
+    integration_platforms:
+      - nrf9160dk_nrf9160_ns
+    platform_allow: nrf9160dk_nrf9160_ns
+    tags: ci_build
+  sample.nrf7002.shell.scan_only_9160:
```
[the remark this answers is not in the document]
Comment (rado17) [reply samples/wifi/shell/sample.yaml]: Done.

**Review by sachinthegreen — APPROVED** [review]

File: boards/arm/nrf7001dk_nrf5340/Kconfig [hunk 7]
```diff
@@ -0,0 +1,85 @@
+# nRF5340 DK board configuration
```
Comment (krish2718) [inline boards/arm/nrf7001dk_nrf5340/Kconfig:1]:
```suggestion
# nRF7001 DK board configuration
```

File: boards/arm/nrf7001dk_nrf5340/Kconfig.board [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# nRF5340 DK NRF5340 board configuration
```
Comment (krish2718) [inline boards/arm/nrf7001dk_nrf5340/Kconfig.board:1]:
```suggestion
# nRF7001 DK NRF5340 board configuration
```

File: samples/nrf9160/modem_shell/sample.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (trantanen) [reply samples/nrf9160/modem_shell/sample.yaml]: I would keep only nrf7002 version for debug and pgps configurations. But my first comment still applies that I don't know if there is a reason to compile everything with all nrf700x variants.

So I would remove these builds from modem_shell sample.yaml:
sample.nrf9160.modem_shell.nrf7000ek_wifi-debug
sample.nrf9160.modem_shell.nrf7001ek_wifi-debug
sample.nrf9160.modem_shell.location_service_ext_pgps_nrf7000ek_wifi
sample.nrf9160.modem_shell.location_service_ext_pgps_nrf7001ek_wifi

File: boards/arm/nrf7001dk_nrf5340/Kconfig.defconfig [hunk 9]
```diff
@@ -0,0 +1,93 @@
+# nRF5340 DK nRF5340 board configuration
```
Comment (krish2718) [inline boards/arm/nrf7001dk_nrf5340/Kconfig.defconfig:1]:
```suggestion
# nRF7001 DK nRF5340 board configuration
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c [hunk 10]
```diff
@@ -168,6 +169,7 @@ enum wifi_nrf_status umac_cmd_init(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx,
 	umac_cmd_data->ant_gain_5g_band1 = ant_gain_5g_band1;
 	umac_cmd_data->ant_gain_5g_band2 = ant_gain_5g_band2;
 	umac_cmd_data->ant_gain_5g_band3 = ant_gain_5g_band3;
+	umac_cmd_data->single_band_only = single_band;
```
Comment (krish2718) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c:172]: The input arg should also be named `single_band_only` and also, the naming is not quite right, we should instead use an enum `supported_bands` with below or at the very least rename to `band_24g_only` (my preference is enum).

```
BAND_24G
BAND_5G
BAND_ALL
```

File: drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c [hunk 11]
```diff
@@ -264,6 +264,11 @@ enum wifi_nrf_status wifi_nrf_fmac_dev_add_zep(struct wifi_nrf_drv_priv_zep *drv
 	struct wifi_nrf_ctx_zep *rpu_ctx_zep = NULL;
 	struct wifi_nrf_fmac_fw_info fw_info;
 	void *rpu_ctx = NULL;
+#if defined(CONFIG_BOARD_NRF7000) || defined(CONFIG_BOARD_NRF7001)
```
Comment (krish2718) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c:267]: see comment about using enum for this, it will be cleaner.

File: boards/shields/nrf7002_ek/Kconfig.shield [hunk 12]
```diff
@@ -4,5 +4,14 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Comment (krish2718) [inline boards/shields/nrf7002_ek/Kconfig.shield:4]: This directory has to be removed as we are using the new directory

File: boards/arm/nrf7001dk_nrf5340/CMakeLists.txt [hunk 13]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA.
```
Comment (krish2718) [inline boards/arm/nrf7001dk_nrf5340/CMakeLists.txt:1]: Can we please modify the commit to log to add that this is based on nrf7002 copy.

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c (hunk 10 above)
Comment (sachinthegreen) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c:172]: I guess we can have a enum, but not have the BANG_5G value in it right now (since we don't support 5 GHz only operation yet).

Comment (krish2718) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c:172]: yeah, we could keep it generic and UMAC can fail if a combination isn't supported either way is fine with me.

Comment (rado17) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c:172]: Considering that on UMAC side, this variable is used just to identify if single band support is enabled or not (umac_cmd_data->single_band_only), I think we should define the enum as:
```
enum single_band_24g {
Enabled 
Disabled
}
```

Comment (krish2718) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c:172]: As UMAC patch is in this PR, we should change it to take enum as well.

File: samples/nrf9160/modem_shell/sample.yaml [hunk 14]
```diff
@@ -117,7 +136,31 @@ tests:
...
+      - nrf9160dk_nrf9160_ns
+    platform_allow: nrf9160dk_nrf9160_ns
+    tags: ci_build
+  sample.nrf9160.modem_shell.location_service_ext_pgps_nrf7001ek_wifi:
+    build_only: true
+    extra_configs:
+      - CONFIG_LOCATION_SERVICE_EXTERNAL=y CONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE=y
+    extra_args: SHIELD=nrf7002ek_nrf7001
+                OVERLAY_CONFIG="overlay-cloud_mqtt.conf;overlay-pgps.conf;overlay-nrf7002ek-wifi-scan-only.conf"
+                DTC_OVERLAY_FILE="nrf9160dk_with_nrf7002ek.overlay"
```
Comment (tokangas) [inline samples/nrf9160/modem_shell/sample.yaml:163]: As suggested by @trantanen in https://github.com/nrfconnect/sdk-nrf/pull/10964#discussion_r1187179501, sample.nrf9160.modem_shell.location_service_ext_pgps_nrf7000ek_wifi and sample.nrf9160.modem_shell.location_service_ext_pgps_nrf7001ek_wifi could be removed. It should be enough to build the default configuration for all three EKs.

Comment (rado17) [reply samples/nrf9160/modem_shell/sample.yaml]: Done

File: boards/arm/nrf7001dk_nrf5340/CMakeLists.txt (hunk 13 above)
Comment (rado17) [reply boards/arm/nrf7001dk_nrf5340/CMakeLists.txt:1]: Done

File: boards/shields/nrf7002_ek/Kconfig.shield (hunk 12 above)
Comment (rado17) [reply boards/shields/nrf7002_ek/Kconfig.shield]: Done

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c (hunk 10 above)
Comment (rado17) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c]: Done

**Review by trantanen — APPROVED** [review]

File: samples/nrf9160/modem_shell/sample.yaml [hunk 15]
```diff
@@ -40,14 +40,33 @@ tests:
     integration_platforms:
       - nrf9160dk_nrf9160_ns
     platform_allow: nrf9160dk_nrf9160_ns
-    extra_args: SHIELD=nrf7002_ek OVERLAY_CONFIG=overlay-nrf7002ek-wifi-scan-only.conf DTC_OVERLAY_FILE="nrf9160dk_with_nrf7002ek.overlay"
+    extra_args: SHIELD=nrf7002ek_nrf7002 OVERLAY_CONFIG=overlay-nrf7002ek-wifi-scan-only.conf
```
Comment (trantanen) [inline samples/nrf9160/modem_shell/sample.yaml:43]: Commenting here to get a conversation but you'll need a change of `SHIELD` variable for `applications/asset_tracker_v2/sample.yaml` also to get CI through.

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fmac_api.h [hunk 16]
```diff
@@ -1203,7 +1203,8 @@ enum wifi_nrf_status wifi_nrf_fmac_dev_init(struct wifi_nrf_fmac_dev_ctx *fmac_d
 					    unsigned char ant_gain_2g,
 					    unsigned char ant_gain_5g_band1,
 					    unsigned char ant_gain_5g_band2,
-					    unsigned char ant_gain_5g_band3);
+					    unsigned char ant_gain_5g_band3,
+						unsigned int op_band);
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fmac_api.h:1207]: Indentation.

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/src/fmac_api.c [hunk 17]
```diff
@@ -213,7 +214,8 @@ static enum wifi_nrf_status wifi_nrf_fmac_fw_init(struct wifi_nrf_fmac_dev_ctx *
 			       ant_gain_2g,
 			       ant_gain_5g_band1,
 			       ant_gain_5g_band2,
-			       ant_gain_5g_band3);
+			       ant_gain_5g_band3,
+				   op_band);
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/src/fmac_api.c:218]: Indentation.

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/src/fmac_api.c [hunk 18]
```diff
@@ -447,7 +450,8 @@ enum wifi_nrf_status wifi_nrf_fmac_dev_init(struct wifi_nrf_fmac_dev_ctx *fmac_d
 				       ant_gain_2g,
 				       ant_gain_5g_band1,
 				       ant_gain_5g_band2,
-				       ant_gain_5g_band3);
+				       ant_gain_5g_band3,
+					   op_band);
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/src/fmac_api.c:454]: Indentation.

File: boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpunet_defconfig [hunk 19]
```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Comment (sachinthegreen) [inline boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpunet_defconfig:1]: `# Copyright (c) 2023 Nordic Semiconductor ASA` missing?

File: boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpuapp_ns_defconfig [hunk 20]
```diff
@@ -0,0 +1,30 @@
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Comment (sachinthegreen) [inline boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpuapp_ns_defconfig:1]: `# Copyright (c) 2023 Nordic Semiconductor ASA` missing?

File: boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpuapp_defconfig [hunk 21]
```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Comment (sachinthegreen) [inline boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpuapp_defconfig:1]: `# Copyright (c) 2023 Nordic Semiconductor ASA` missing?

File: boards/arm/nrf7001dk_nrf5340/board.cmake [hunk 22]
```diff
@@ -0,0 +1,20 @@
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Comment (sachinthegreen) [inline boards/arm/nrf7001dk_nrf5340/board.cmake:1]: `# Copyright (c) 2023 Nordic Semiconductor ASA` missing?

**Review by tokangas — APPROVED** [review]

**Review by sachinthegreen — CHANGES_REQUESTED** [review]
In general, instead of having multiple copies of the same file, please try to have a single file called nrf700x_* if the contents are the same for all variants (or even if a single file is possible using conditional statements for different variants)

File: samples/nrf9160/modem_shell/sample.yaml (hunk 15 above)
Comment (rado17) [reply samples/nrf9160/modem_shell/sample.yaml:43]: Done

> In general, instead of having multiple copies of the same file, please try to have a single file called nrf700x_* if the contents are the same for all variants (or even if a single file is possible using conditional statements for different variants)

Having a file of nrf700x format is possible for only dts bindings. But setting a generic "compatible" string in dts bindings is interfering with the checks we have added for excluding supplicant support for nrf7000 device. So, not making this change for now.

**Comment by rado17** [comment]
> In general, instead of having multiple copies of the same file, please try to have a single file called nrf700x_* if the contents are the same for all variants (or even if a single file is possible using conditional statements for different variants)

Having a file of nrf700x format is possible for only dts bindings. But setting a generic "compatible" string in dts bindings is interfering with the checks we have added for excluding supplicant support for nrf7000 device. So, not making this change for now.

File: boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpuapp_ns_defconfig (hunk 20 above)
Comment (rado17) [reply boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpuapp_ns_defconfig:2]: Done

File: boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpuapp_defconfig (hunk 21 above)
Comment (rado17) [reply boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpuapp_defconfig:2]: Done

File: boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpunet_defconfig (hunk 19 above)
Comment (rado17) [reply boards/arm/nrf7001dk_nrf5340/nrf7001dk_nrf5340_cpunet_defconfig:2]: Done

File: boards/arm/nrf7001dk_nrf5340/board.cmake (hunk 22 above)
Comment (rado17) [reply boards/arm/nrf7001dk_nrf5340/board.cmake:2]: Done

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c [hunk 23]
```diff
@@ -93,7 +93,8 @@ enum wifi_nrf_status umac_cmd_init(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx,
 				   unsigned char ant_gain_2g,
 				   unsigned char ant_gain_5g_band1,
 				   unsigned char ant_gain_5g_band2,
-				   unsigned char ant_gain_5g_band3)
+				   unsigned char ant_gain_5g_band3,
+				   unsigned int op_band)
```
Comment (krish2718) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/src/cmd.c:97]:
```suggestion
				   enum op_band op_band)
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/src/fmac_api.c (hunk 18 above)
Comment (rado17) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/src/fmac_api.c]: Done

File: samples/nrf9160/modem_shell/sample.yaml (hunk 15 above)
Comment (rado17) [reply samples/nrf9160/modem_shell/sample.yaml:43]: Done

File: drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c [hunk 24]
```diff
@@ -264,6 +264,11 @@ enum wifi_nrf_status wifi_nrf_fmac_dev_add_zep(struct wifi_nrf_drv_priv_zep *drv
...
+	unsigned int op_band = BAND_24G;
```
Comment (krish2718) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c:268]:
```suggestion
	enum op_band op_band = BAND_24G;
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fmac_api.h (hunk 16 above)
Comment (rado17) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fmac_api.h]: Done

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/src/fmac_api.c (hunk 17 above)
Comment (rado17) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/src/fmac_api.c]: Done

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h [hunk 25]
```diff
@@ -568,14 +568,62 @@ struct nrf_wifi_sys_params {
...
 /**
  * struct nrf_wifi_cmd_sys_init - Initialize UMAC
  * @sys_head: umac header, see &nrf_wifi_sys_head
  * @wdev_id : id of the interface.
  * @sys_params: iftype, mac address, see nrf_wifi_sys_params
  * @rx_buf_pools: LMAC Rx buffs pool params, see struct rx_buf_pool_params
  * @data_config_params: Data configuration params, see struct nrf_wifi_data_config_params
- * @tcp_ip_checksum_offload: 0: Native checksum, 1: Offload checksum
+ * @tcp_ip_checksum_offload: 0:umac checksum disable 1: umac checksum enable
+ * @op_band: operating band see enum op_band
```
Comment (krish2718) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h:626]:
```suggestion
 * @op_band: operating band see &enum op_band
```

**Review by krish2718 — COMMENTED** [review]
> shields: nrf7002_ek: Modify nrf7002 EK shield.

The commit log should be updated to reflect DK changes as well, or use a new commit.

File: samples/wifi/radio_test/src/nrf_wifi_radio_test_shell.c [hunk 26]
```diff
@@ -149,52 +149,80 @@ static bool check_valid_data_rate(const struct shell *shell,
...
+{
+	bool ret = false;
+
+	ret = check_valid_chan_2g(chan_num);
+
+	if (ret) {
+		goto out;
+	}
+
+#ifndef CONFIG_BOARD_NRF7001
```
Comment (krish2718) [inline samples/wifi/radio_test/src/nrf_wifi_radio_test_shell.c:217]: I guess radio test is also applicable for NRF700. So, shouldn't this be `#ifdefCONFIG_BOARD_NRF7002DK_NRF5340_CPUAPP` ?

File: boards/shields/nrf7002ek_nrf7000/Kconfig.shield [hunk 27]
```diff
@@ -0,0 +1,24 @@
...
+
+config NRF7000_ON_QSPI
+	def_bool $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF700X_QSPI))
+	depends on !WPA_SUPP && SHIELD_NRF7002EK_NRF7000
+
+config NRF7000_ON_SPI
+	def_bool $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF700X_SPI))
+	depends on !WPA_SUPP && SHIELD_NRF7002EK_NRF7000
+
+config BOARD_NRF7000
```
Comment (krish2718) [inline boards/shields/nrf7002ek_nrf7000/Kconfig.shield:22]: Why not use the `SHIELD_NRF7002EK_NRF7000` directly? Also this doesn't conform to the other BOARD_* defines in NRF.

File: drivers/wifi/nrf700x/Kconfig.nrf700x [hunk 28]
```diff
@@ -43,11 +40,11 @@ config NET_CONFIG_INIT_TIMEOUT
 	default 0 if NET_CONFIG_SETTINGS
 
 config NRF700X_ON_QSPI
-	def_bool $(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF7002_QSPI))
+	def_bool NRF7000_ON_QSPI
```
Comment (krish2718) [inline drivers/wifi/nrf700x/Kconfig.nrf700x]: Though ultimately its what we want, this commit still breaks the build, it should be `NRF7002_ON_QSPI || NRF7000_ON_QSPI`, NRF7002 should work with all commits.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c [hunk 29]
```diff
@@ -264,6 +264,11 @@ enum wifi_nrf_status wifi_nrf_fmac_dev_add_zep(struct wifi_nrf_drv_priv_zep *drv
 	struct wifi_nrf_ctx_zep *rpu_ctx_zep = NULL;
 	struct wifi_nrf_fmac_fw_info fw_info;
 	void *rpu_ctx = NULL;
+#if defined(CONFIG_BOARD_NRF7000) || defined(CONFIG_BOARD_NRF7001)
+	enum op_band op_band = BAND_24G;
```
Comment (krish2718) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c:268]: How will this work with OTP programming? We need to document the requirements somewhere?

**Review by krish2718 — COMMENTED** [review]
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/working_with_nrf/nrf70/features.html#supported-boards This needs to be extended  also I guess we should add entries similar to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/working_with_nrf/nrf70/gs.html they could be minimal just explain new board specific stuff and refer nrf7002 DK, I guess this is needed at least for 7001 DK . We haven't added nRF7002 EK in docs intentionally as that time support for it was only internal, but now I guess we should add that as well.

TL;DR, we need to add doc about each variants what it means and its features.

File: boards/shields/nrf7002ek_nrf7000/Kconfig.shield (hunk 27 above)
Comment (rado17) [reply boards/shields/nrf7002ek_nrf7000/Kconfig.shield:22]: BOARD_NRF7000 is to facilitate board-specific checks irrespective of it being a DK or an EK. Thats why SHIELD  config is being translated into a board config.

**Comment by krish2718** [comment]
> https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/working_with_nrf/nrf70/features.html#supported-boards This needs to be extended also I guess we should add entries similar to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/working_with_nrf/nrf70/gs.html they could be minimal just explain new board specific stuff and refer nrf7002 DK, I guess this is needed at least for 7001 DK . We haven't added nRF7002 EK in docs intentionally as that time support for it was only internal, but now I guess we should add that as well.
> 
> TL;DR, we need to add doc about each variants what it means and its features.

https://github.com/nrfconnect/sdk-nrf/pull/11054/files this is taking care of here.

File: samples/wifi/sr_coex/README.rst [hunk 30]
```diff
@@ -211,6 +211,22 @@ Use this command for Wi-Fi throughput only, Bluetooth LE throughput only, or con
 
 Use this command for concurrent Wi-Fi and Bluetooth LE throughput with coexistence enabled test.
 
+Change the build target as given below for the nRF7001 DK, nRF7002 EK and nRF7001 EK.
```
Comment (kittydepa) [inline samples/wifi/sr_coex/README.rst:214]:
```suggestion
Change the build target as given below for the nRF7001 DK, nRF7002 EK and nRF7001 EK.

```

File: samples/wifi/sr_coex/README.rst [hunk 31]
```diff
@@ -211,6 +211,22 @@ Use this command for Wi-Fi throughput only, Bluetooth LE throughput only, or con
...
+Build target for nRF7001 DK:
```
Comment (kittydepa) [inline samples/wifi/sr_coex/README.rst:215]:
```suggestion
Build target for nRF7001 DK:

```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 32]
```diff
@@ -158,6 +158,15 @@ nRF IEEE 802.15.4 radio driver
 Wi-Fi
 -----
 
+* Added:
+
+  * Support for nRF7000 EK
```
Comment (kittydepa) [inline doc/nrf/releases/release-notes-changelog.rst:163]:
```suggestion
  * Support for nRF7000 EK.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 33]
```diff
@@ -158,6 +158,15 @@ nRF IEEE 802.15.4 radio driver
...
+  * Support for nRF7001 EK and nRF7001 DK.
+
+* Updated:
+
+  * The shield for nRF7002 EK (nrf7002_ek -> nrf7002ek_nrf7002).
```
Comment (kittydepa) [inline doc/nrf/releases/release-notes-changelog.rst:168]:
```suggestion
  * The shield for nRF7002 EK (``nrf7002_ek`` -> ``nrf7002ek_nrf7002``).
```

File: samples/wifi/sr_coex/README.rst [hunk 34]
```diff
@@ -211,6 +211,22 @@ Use this command for Wi-Fi throughput only, Bluetooth LE throughput only, or con
...
+.. code-block:: console
+	nrf7001dk_nrf5340_cpuapp
+Build target for nRF7002 EK and nRF7001 EK:
+.. code-block:: console
+	nrf5340dk_nrf5340_cpuapp
+
+Add the following SHIELD options for the nRF7002 EK and nRF7001 EK.
+For nRF7002 EK:
+.. code-block:: console
+	-DSHIELD=nrf7002ek_nrf7002 -Dhci_rpmsg_SHIELD=nrf7002ek_nrf7002_coex
```
Comment (kittydepa) [inline samples/wifi/sr_coex/README.rst:225]: Please add a blank line before and after the ``.. code-block:: console `` entries. Like so:

> For nRF7001 EK:
>
>  .. code-block:: console
>  (indented code here)
>
> The generated HEX file...

**Review by kittydepa — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf700x/Kconfig.nrf700x (hunk 28 above)
Comment (rado17) [reply drivers/wifi/nrf700x/Kconfig.nrf700x]: Done.

File: samples/wifi/sr_coex/README.rst (hunk 30 above)
Comment (rado17) [reply samples/wifi/sr_coex/README.rst:214]: Done

File: samples/wifi/sr_coex/README.rst (hunk 31 above)
Comment (rado17) [reply samples/wifi/sr_coex/README.rst:216]: Done

File: doc/nrf/releases/release-notes-changelog.rst (hunk 32 above)
Comment (rado17) [reply doc/nrf/releases/release-notes-changelog.rst]: Done

File: doc/nrf/releases/release-notes-changelog.rst (hunk 33 above)
Comment (rado17) [reply doc/nrf/releases/release-notes-changelog.rst]: Done

File: samples/wifi/sr_coex/README.rst (hunk 34 above)
Comment (rado17) [reply samples/wifi/sr_coex/README.rst]: Done
